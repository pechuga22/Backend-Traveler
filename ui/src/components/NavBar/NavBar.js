import React from 'react'
import { Link } from 'react-router-dom'
import { Menu, Header, Icon, Segment, Image } from 'semantic-ui-react'

import isLogin from '../../utils/index'
import Avatar from '../../static/avatar-traveler.png'
import './NavBar.less'

export default function NavBar({ searchFocus }) {
  return (
    <Menu widths={3} fixed='bottom' className='main navbar'>
      <Menu.Item data-testid='item'>
        <Segment basic>
          <Icon name='search' size='large' onClick={searchFocus} />
          <Header as='h5'>EXPLORE</Header>
        </Segment>
      </Menu.Item>

      <Menu.Item data-testid='item'>
        <Link to='/favorites'>
        <Segment basic>
          <Icon name='heart outline' size='large' />
          <Header as='h5'>SAVED</Header>
        </Segment>
        </Link>
      </Menu.Item>

      <Menu.Item className='login' data-testid='item'>
        <Link to='/register'>
          <Segment basic>
            {  isLogin() ?  <Image src ={Avatar}  circular size='mini'/> : <Icon className='user' name='user' size='large' />}
            <Header className='profile'as='h5'>PROFILE</Header>
          </Segment>
        </Link>
      </Menu.Item>
    </Menu>
  )
}
