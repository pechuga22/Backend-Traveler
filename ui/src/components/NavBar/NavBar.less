@import '../../variables';

div.ui.menu.item.bottom.fixed.main.navbar {
  height: 93px;
  border-radius: 50px !important;
  box-shadow: 0 -3px 20px 0 rgba(0, 0, 0, 0.16);
  backdrop-filter: blur(10px);
  opacity: 0.9;

  :before {
    background: none;
  }

  .item:hover {
    .ui.basic.segment,
    .header:not(h6) {
      color: @salmonPink;
    }
  }

  .ui.basic.segment {
    color: @greySecondary;
    img {
      margin: 0 auto;
      padding-bottom: 0px;
      position: relative;
      top: 1px;
    }
    .profile {
      bottom: 5px;
      margin-bottom: 4px;
      padding-top: 0px;
      position: relative;
    }
    .user {
      margin-bottom: 13px;
      padding-top: 8px;
    }

  }




  .header:not(h6) {
    margin-top: 0.8rem;
    color: @greySecondary;
  }
}
