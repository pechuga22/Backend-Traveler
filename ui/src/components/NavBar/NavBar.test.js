import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import NavBar from './NavBar'

describe('./components/NavBar/NavBar.js', () => {
  describe('.render()', () => {
    const component = shallow(<NavBar />)

    it('renders and matches our snapshot', () => {
      const component = shallow(<NavBar />)
      expect(toJson(component)).toMatchSnapshot()
    })

    it('find the login buttons in NavBar', () => {
      const wrapper = component.find("[data-testid='item']")
      expect(wrapper.length).toBe(3)
    })
  })
})
