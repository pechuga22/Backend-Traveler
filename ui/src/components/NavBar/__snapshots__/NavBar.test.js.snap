// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`./components/NavBar/NavBar.js .render() renders and matches our snapshot 1`] = `
<Menu
  className="main navbar"
  fixed="bottom"
  widths={3}
>
  <MenuItem
    data-testid="item"
  >
    <Segment
      basic={true}
    >
      <Icon
        as="i"
        name="search"
        size="large"
      />
      <Header
        as="h5"
      >
        EXPLORE
      </Header>
    </Segment>
  </MenuItem>
  <MenuItem
    data-testid="item"
  >
    <Link
      to="/favorites"
    >
      <Segment
        basic={true}
      >
        <Icon
          as="i"
          name="heart outline"
          size="large"
        />
        <Header
          as="h5"
        >
          SAVED
        </Header>
      </Segment>
    </Link>
  </MenuItem>
  <MenuItem
    className="login"
    data-testid="item"
  >
    <Link
      to="/register"
    >
      <Segment
        basic={true}
      >
        <Icon
          as="i"
          className="user"
          name="user"
          size="large"
        />
        <Header
          as="h5"
          className="profile"
        >
          PROFILE
        </Header>
      </Segment>
    </Link>
  </MenuItem>
</Menu>
`;
